packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -54,27 +54,43 @@ enum AliasState {
 class DataStore implements IDataStore {
     private aliasState: AliasState = AliasState.None;
     private alias: string | undefined;
+    private aliasResult: Promise<AliasResult> | undefined;
 
     async trySetAlias(alias: string): Promise<AliasResult> {
         if (alias.includes("/")) {
             throw new UsageError(`The alias cannot contain slashes: '${alias}'`);
         }
 
         switch (this.aliasState) {
-            // If we're already aliasing, do nothing
-            case AliasState.Aliasing:
-                return "Aliasing";
+            // If we're already aliasing, check if it's for the same value and return
+            // the stored promise, otherwise return 'AlreadyAliased'
+            case AliasState.Aliasing: {
+                assert(this.aliasResult !== undefined, "There should be a cached promise of in-progress aliasing");
+                const pendingAliasResult = await this.aliasResult;
+                if (pendingAliasResult !== "Success") {
```

Comment:
is this check correct? It feels like it does not matter if previous succussed or failed, we know that some alias was assigned, and not we can return result based on return statement below.

Even if prior request from same machine failed, it might have been for different name, and thus it's not that interesting.

Reply:
@vladsud I believe it is. This catches the following case: https://github.com/microsoft/FluidFramework/pull/10303/files#diff-5f8ba9e0e468e76737c8e8be3c120c57d1901874c686f7e6ce682711ad2c3fb7R348 so when `trySetAlias` is called on the same datastore with a different alias parameter while it's already aliased.

Reply:
Updated + unit test.